specification/hybridnetwork/resource-manager/Microsoft.HybridNetwork/stable/2024-04-15/common.json

```diff
@@ -0,0 +1,611 @@
+    "description": "The definitions in this swagger specification will be used to manage the Hybrid Network resources.",
+    "version": "2024-04-15"
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "security": [
+    {
+      "azure_auth": [
+        "user_impersonation"
+      ]
+    }
+  ],
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow.",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {},
+  "definitions": {
+    "ReferencedResource": {
+      "type": "object",
+      "properties": {
+        "id": {
+          "type": "string",
+          "description": "Resource ID."
+        }
+      },
+      "description": "Reference to another resource.",
+      "x-ms-azure-resource": true
+    },
+    "CustomLocationResourceId": {
+      "type": "object",
+      "properties": {
+        "id": {
+          "type": "string",
+          "pattern": "^/[sS][uU][bB][sS][cC][rR][iI][pP][tT][iI][oO][nN][sS]/[^/?#]+/[rR][eE][sS][oO][uU][rR][cC][eE][gG][rR][oO][uU][pP][sS]/[^/?#]+/[pP][rR][oO][vV][iI][dD][eE][rR][sS]/[mM][iI][cC][rR][oO][sS][oO][fF][tT]\\.[eE][xX][tT][eE][nN][dD][eE][dD][lL][oO][cC][aA][tT][iI][oO][nN]/[cC][uU][sS][tT][oO][mM][lL][oO][cC][aA][tT][iI][oO][nN][sS]/[^/?#]+$",
+          "description": "Azure ARC custom location resource ID."
+        }
+      },
+      "description": "Reference to an Azure ARC custom location resource.",
+      "x-ms-azure-resource": true
+    },
+    "ProvisioningState": {
+      "type": "string",
+      "readOnly": true,
+      "description": "The current provisioning state.",
+      "enum": [
+        "Unknown",
+        "Succeeded",
+        "Accepted",
+        "Deleting",
+        "Failed",
+        "Canceled",
+        "Deleted",
+        "Converging"
+      ],
+      "x-ms-enum": {
+        "name": "ProvisioningState",
+        "modelAsString": true
+      }
+    },
+    "NetworkFunctionType": {
+      "type": "string",
+      "description": "The network function type.",
+      "enum": [
+        "Unknown",
+        "VirtualNetworkFunction",
+        "ContainerizedNetworkFunction"
+      ],
+      "x-ms-mutability": [
+        "read",
+        "create"
+      ],
+      "x-ms-enum": {
+        "name": "NetworkFunctionType",
+        "modelAsString": true
+      }
+    },
+    "NFVIType": {
+      "type": "string",
+      "description": "The NFVI type.",
+      "enum": [
+        "Unknown",
+        "AzureArcKubernetes",
+        "AzureCore",
+        "AzureOperatorNexus"
+      ],
+      "x-ms-enum": {
+        "name": "NFVIType",
+        "modelAsString": true
+      }
+    },
+    "backingResourcePublicNetworkAccess": {
```

Comment:
>backingResourcePublicNetworkAccess [](http://example.com/codeflow?start=5&length=34)

Even though this is defined as enum, it's acting like boolean because it's hard to extend the values given the current values (Enabled and Disabled).

Try renaming it to something like `backingResourceNetworkAccess`. Then you can values such as `Public`, `Private`, `Disallowed`, etc.

Reply:
The intent of this property is to only disable public access or not, customers have already started using the property the current way so we would like to leave it as is

Reply:
I'd like to put the reasoning behind this here. Please acknowledge that you will apply it in your future API design.

_ARM recommends enums over booleans for future proof APIs. Standard guidance is: replace boolean/switch properties with a more meaningful enum whenever possible.

A boolean will forever have two valid values (true or false). A string enum type is always preferred. Also, properties should always provide better values just than True and False. For example two switches "isTypeA" and "isTypeB" should be replaced with one enum "type": [A, B, DefaultType]. Enums are always a more flexible and future proof option because they allow additional values to be added in the future in a non-breaking way, e.g. [Enabled, Disabled, Suspended, Deallocated]. 

Note: do NOT define a 'boolean enum' with two values 'True and False'. This might be easier to 'extend' in terms of types, but semantically its confusing, and no better than a boolean._

Reply:
sure will keep this in mind going forward